substrate/client/keystore/src/local.rs

```diff
@@ -370,7 +393,23 @@ impl Keystore for LocalKeystore {
 			key_type: KeyTypeId,
 			seed: Option<&str>,
 		) -> std::result::Result<ecdsa_bls381::Public, TraitError> {
-			self.generate_new::<ecdsa_bls381::Pair>(key_type, seed)
+			let pubkey = self.generate_new::<ecdsa_bls381::Pair>(key_type, seed)?;
+
+			let s = self
+				.0
+				.read()
+				.additional
+				.get(&(key_type, pubkey.to_vec()))
+				.map(|s| s.to_string())
+				.expect("Can retrieve seed");
+
+			// This is done to give the keystore access to individual keys, this is necessary to avoid
+			// unnecessary host functions for paired keys and re-use host functions implemented for each
+			// element of the pair.
+			self.generate_new::<ecdsa::Pair>(sp_core::testing::ECDSA, Some(&*s)).expect("seed slice is valid");
+			self.generate_new::<bls381::Pair>(sp_core::testing::BLS381, Some(&*s)).expect("seed slice is valid");
```

Comment:
You are using here the `testing::KeyTypes` which would not be used in production. It is very fishy what you are doing here. Can you please revert these changes? Or the code using this should be fixed.